code/modules/mob/living/simple_animal/hostile/venus_human_trap.dm

```diff
@@ -62,75 +84,94 @@
+
+/mob/living/simple_animal/hostile/venus_human_trap/Life()
+	. = ..()
+	pull_vines()
+	
+/mob/living/simple_animal/hostile/venus_human_trap/AttackingTarget()
+	. = ..()
+	if(isliving(target))
+		var/mob/living/L = target
+		if(L.stat != DEAD)
+			adjustHealth(-maxHealth * 0.1)
 
 /mob/living/simple_animal/hostile/venus_human_trap/OpenFire(atom/the_target)
+	for(var/datum/beam/B in vines)
+		if(B.target == the_target)
+			pull_vines()
+			ranged_cooldown = world.time + (ranged_cooldown_time * 0.5)
+			return
+	if(get_dist(src,the_target) > vine_grab_distance || vines.len == max_vines)
+		return
 	for(var/turf/T in getline(src,target))
 		if (T.density)
 			return
 		for(var/obj/O in T)
 			if(O.density)
 				return
-	var/dist = get_dist(src,the_target)
-	Beam(the_target, "vine", time=dist*2, maxdistance=dist+2, beam_type=/obj/effect/ebeam/vine)
-	the_target.attack_animal(src)
+	vines += Beam(the_target, "vine", time=INFINITY, maxdistance = vine_grab_distance, beam_type=/obj/effect/ebeam/vine)
+	if(isliving(the_target))
+		var/mob/living/L = the_target
+		L.Paralyze(20)
+	ranged_cooldown = world.time + ranged_cooldown_time
 
+/mob/living/simple_animal/hostile/venus_human_trap/Login()
+	. = ..()
+	to_chat(src, "<span class='boldwarning'>You a venus human trap!  Protect the kudzu at all costs, and feast on those who oppose you!</span>")
 
-/mob/living/simple_animal/hostile/venus_human_trap/CanAttack(atom/the_target)
+/mob/living/simple_animal/hostile/venus_human_trap/attack_ghost(mob/user)
 	. = ..()
 	if(.)
-		if(the_target in grasping)
-			return 0
+		return
+	humanize_plant(user)
+
+/**
+  * Sets a ghost to control the plant if the plant is eligible
+  *
+  * Asks the interacting ghost if they would like to control the plant.
+  * If they answer yes, and another ghost hasn't taken control, sets the ghost to control the plant.
+  * Arguments:
+  * * mob/user - The ghost to possibly control the plant
+  */
+/mob/living/simple_animal/hostile/venus_human_trap/proc/humanize_plant(mob/user)
+	if(key || !playable_plant || stat)
+		return
+	var/plant_ask = alert("Become a venus human trap?", "Are you reverse vegan?", "Yes", "No")
+	if(plant_ask == "No" || QDELETED(src))
+		return
+	if(key)
+		to_chat(user, "<span class='warning'>Someone else already took this plant!</span>")
+		return
+	key = user.key
+	log_game("[key_name(src)] took control of [name].")
+
+/**
+  * Manages how the vines should affect the things they're attached to.
+  *
+  * Pulls all movable targets of the vines closer to the plant
+  * If the target is on the same tile as the plant, destroy the vine
+  * Removes any QDELETED vines from the vines list.
+  */
+/mob/living/simple_animal/hostile/venus_human_trap/proc/pull_vines()
+	for(var/datum/beam/B in vines)
+		if(istype(B.target, /atom/movable))
+			var/atom/movable/AM = B.target
+			if(!AM.anchored)
+				step(AM,get_dir(AM,src))
+		if(get_dist(src,B.target) == 0)
+			B.End()
+		if(QDELETED(B))
+			vines -= B
```

Comment:
Handle this in the beam's `Destroy()` or better yet register for a destruction signal.

Reply:
Could you explain what you mean by a destruction signal?

Reply:
Do you know how signals work? Poke me on #coding-general if you want it explained. The signal we'd be listening to would be `COMSIG_PARENT_QDELETING` in this case.